### PR TITLE
clarify TLS framing

### PR DESCRIPTION
**Pull request by tatianab**
Proposal to fix Issue #57

### CODE REVIEW

File: draft-ietf-tls-exported-authenticator.md [hunk 1]
```diff
@@ -131,8 +131,9 @@ which uses the same structure as CertificateRequest.  These messages
 structures are used even if the TLS connection protocol is TLS 1.2.
 
 The CertificateRequest and ClientCertificateRequest messages are used to define the
-parameters in a request for an authenticator.  These messages do not include any
-TLS framing and are not encrypted with a handshake key.
+parameters in a request for an authenticator.  These messages are encoded as 
+Handshake Messages but do not include any TLS record layer framing, and are not
+encrypted with a handshake key.
```
Comment (chris-wood) [inline draft-ietf-tls-exported-authenticator.md]:
```suggestion
parameters in a request for an authenticator.  These are encoded as TLS handshake messages, including 
length and type fields. They do not include any TLS record layer framing and are not encrypted with a handshake key.
```

File: draft-ietf-tls-exported-authenticator.md [hunk 2]
```diff
@@ -215,13 +216,14 @@ used with cipher suites that do not use the TLS PRF and have not defined
...
-accepted on TLS 1.2 connectons that did not negotiate
+accepted on TLS 1.2 connections that did not negotiate
 the extended master secret {{!RFC7627}}.
 
 ## Authenticator Construction
 
 An authenticator is formed from the concatenation of TLS 1.3 {{!TLS13}}
-Certificate, CertificateVerify, and Finished messages.
+Certificate, CertificateVerify, and Finished messages. These messages are
+encoded as Handshake Messages, including the length and type fields.
```
Comment (chris-wood) [inline draft-ietf-tls-exported-authenticator.md]:
```suggestion
encoded as TLS handshake messages, including length and type fields.
```

File: draft-ietf-tls-exported-authenticator.md [hunk 3]
```diff
@@ -356,7 +358,8 @@ message sent without a Certificate or CertificateVerify. This message is an
 HMAC over the hashed authenticator transcript with a Certificate message
 containing no CertificateEntries and the CertificateVerify message omitted.
 The HMAC is computed using the authenticator hash, using the Finished MAC Key as a key.
-This message does not include any TLS framing.
+This message is encoded as a Handshake Message but does not include any TLS record
+layer framing.
```
Comment (chris-wood) [inline draft-ietf-tls-exported-authenticator.md]:
```suggestion
This message is encoded as a TLS handshake message, including length and type field. It does not include TLS record layer framing.
```

**Comment by chris-wood** [comment]
cc @grittygrease

File: draft-ietf-tls-exported-authenticator.md [hunk 4]
```diff
@@ -131,8 +131,8 @@ which uses the same structure as CertificateRequest.  These messages
 structures are used even if the TLS connection protocol is TLS 1.2.
 
 The CertificateRequest and ClientCertificateRequest messages are used to define the
-parameters in a request for an authenticator.  These messages do not include any
-TLS framing and are not encrypted with a handshake key.
+parameters in a request for an authenticator.  These are encoded as TLS handshake messages, including 
+length and type fields. They do not include any TLS record layer framing and are not encrypted with a handshake key.
```
Comment (Lekensteyn) [inline draft-ietf-tls-exported-authenticator.md:135]: To match the wire format, I'd suggest swapping "length" and "type".

**Comment by Lekensteyn** [comment]
The Introduction limits this to TLS 1.2 and later, but DTLS is similar. Do you want to take that into account now, or ignore it?

DTLS has additional fields to support fragmentation:
https://tools.ietf.org/html/rfc6347#section-4.2.2
https://tools.ietf.org/html/draft-ietf-tls-dtls13-37#section-5.2

**Review by grittygrease — APPROVED** [review]

**Comment by grittygrease** [comment]
> The Introduction limits this to TLS 1.2 and later, but DTLS is similar. Do you want to take that into account now, or ignore it?
> 
> DTLS has additional fields to support fragmentation:
> https://tools.ietf.org/html/rfc6347#section-4.2.2
> https://tools.ietf.org/html/draft-ietf-tls-dtls13-37#section-5.2

Exported authentication are meant to be embedded in a higher-layer protocol, so there's no need to support fragmentation.

**Pull request merged**